mkosi/qemu.py

```diff
@@ -537,7 +537,7 @@ def run_qemu(args: MkosiArgs, config: MkosiConfig, qemu_device_fds: Mapping[Qemu
 
     machine = f"type={config.architecture.default_qemu_machine()}"
     if firmware == QemuFirmware.uefi:
-        machine += f",smm={'on' if ovmf_supports_sb else 'off'}"
+        machine += f",smm={'on' if ovmf_supports_sb and config.architecture.supports_smm() else 'off'}"
```

Comment:
Wouldn't it make sense to move the `config.architecture.supports_smm()` calls into the places where `find_ovmf_firmware` returns `True` for the second argument?

Reply:
I don't think so, these are two different concepts